pkg/ui/workspaces/db-console/src/views/jobs/jobDetails.tsx

```diff
@@ -2,62 +2,35 @@
 //
 // Use of this software is governed by the CockroachDB Software License
 // included in the /LICENSE file.
-import {
-  JobDetails,
-  JobDetailsStateProps,
-  selectID,
-  api as clusterUiApi,
-} from "@cockroachlabs/cluster-ui";
-import long from "long";
-import { connect } from "react-redux";
+import { api as clusterUiApi, JobDetailsV2 } from "@cockroachlabs/cluster-ui";
+import React from "react";
+import { useDispatch, useStore } from "react-redux";
+import { StaticContext } from "react-router";
 import { RouteComponentProps, withRouter } from "react-router-dom";
 
-import { collectExecutionDetailsAction } from "oss/src/redux/jobs/jobsActions";
-import {
-  createSelectorForKeyedCachedDataField,
-  refreshListExecutionDetailFiles,
-  refreshJob,
-  refreshUserSQLRoles,
-} from "src/redux/apiReducers";
-import { AdminUIState, AppDispatch } from "src/redux/state";
+import { refreshUserSQLRoles } from "src/redux/apiReducers";
 import { selectHasAdminRole } from "src/redux/user";
-import { ListJobProfilerExecutionDetailsResponseMessage } from "src/util/api";
-
-const selectJob = createSelectorForKeyedCachedDataField("job", selectID);
-const selectExecutionDetailFiles =
-  createSelectorForKeyedCachedDataField<ListJobProfilerExecutionDetailsResponseMessage>(
-    "jobProfiler",
-    selectID,
-  );
-const mapStateToProps = (
-  state: AdminUIState,
-  props: RouteComponentProps,
-): JobDetailsStateProps => {
-  const jobID = selectID(state, props);
-  return {
-    jobRequest: selectJob(state, props),
-    jobProfilerExecutionDetailFilesResponse: selectExecutionDetailFiles(
-      state,
-      props,
-    ),
-    jobProfilerLastUpdated: state.cachedData.jobProfiler[jobID]?.setAt,
-    jobProfilerDataIsValid: state.cachedData.jobProfiler[jobID]?.valid,
-    onDownloadExecutionFileClicked: clusterUiApi.getExecutionDetailFile,
-    hasAdminRole: selectHasAdminRole(state),
-  };
-};
-
-const mapDispatchToProps = {
-  refreshJob,
-  refreshExecutionDetailFiles: refreshListExecutionDetailFiles,
-  onRequestExecutionDetails: (jobID: long) => {
-    return (dispatch: AppDispatch) => {
-      dispatch(collectExecutionDetailsAction({ job_id: jobID }));
-    };
-  },
-  refreshUserSQLRoles: refreshUserSQLRoles,
-};
+import { listExecutionDetailFiles } from "src/util/api";
 
 export default withRouter(
-  connect(mapStateToProps, mapDispatchToProps)(JobDetails),
+  (
+    props: React.PropsWithChildren<
+      RouteComponentProps<any, StaticContext, unknown>
+    >,
+  ) => {
+    const store = useStore();
+    const dispatch = useDispatch();
+    return (
+      <JobDetailsV2
+        {...props}
+        refreshUserSQLRoles={() =>
+          refreshUserSQLRoles()(dispatch, store.getState, undefined)
+        }
```

Comment:
Not sure if there's a better way to do this since `refreshUserSQLRoles` comes from our `CachedDataReducer`, so if anyone's got any better ideas, I would love to hear them.

Reply:
I think this is fine for now. Eventually it'd be nice to use swr for all of these reqs.